features/include/pcl/features/impl/scurv.hpp

```diff
@@ -0,0 +1,730 @@
+#include <math.h>
+#include <numeric>
+#include <algorithm>
+#include <functional>
+#include <typeinfo>
+#include <stdio.h>
+#include <stdlib.h>
+#include <utility>
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointInT, typename PointNT> double
+pcl::SCurVEstimation<PointInT, PointNT>::getNormalizedValue (double x, double low, double high)
+{
+  return (x - low) / (high - low);
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointInT, typename PointNT> double
+pcl::SCurVEstimation<PointInT, PointNT>::getHermiteDerivativeInterpolation(HermitePoint point0, HermitePoint point1, double xi)
+{
+  double h, delta, del0, del1, c2, c3, x;
+  h = point1.x - point0.x;
+  delta = ( point1.f - point0.f ) / h;
+  del0 = ( point0.d - delta ) / h;
+  del1 = ( point1.d - delta ) / h;
+  c2 = -( del0 + del0 + del1 );
+  c3 = ( del0 + del1 ) / h;
+  x = xi - point0.x;
+  return point0.f + x * ( point0.d + x * ( c2 + x * c3 ) );
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointInT, typename PointNT> void
+pcl::SCurVEstimation<PointInT, PointNT>::normalizeScale (pcl::PointCloud<pcl::PointNormal> &cloud, int min_range, int max_range)
+{
+  Eigen::Vector4f min_pt, max_pt;
+  Eigen::Vector3f min_coeff, max_coeff;
+  double minimum, maximum;
+
+  pcl::getMinMax3D(cloud, min_pt, max_pt);
+  min_coeff << min_pt(0), min_pt(1), min_pt(2);
+  max_coeff << max_pt(0), max_pt(1), max_pt(2);
+  minimum = min_coeff.minCoeff();
+  maximum = max_coeff.maxCoeff();
+
+  double factor = std::abs(minimum - maximum) / (max_range - min_range);
+  for (size_t k = 0; k < cloud.points.size(); k++)
+  {
+    cloud.points[k].x = (double)cloud.points[k].x / factor;
+    cloud.points[k].y = (double)cloud.points[k].y / factor;
+    cloud.points[k].z = (double)cloud.points[k].z / factor;
+  }
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointInT, typename PointNT> double
+pcl::SCurVEstimation<PointInT, PointNT>::signMultiplied (double arg1, double arg2)
+{
+  double value = NAN;
+
+  if ( std::fabs(arg1 - 0.0) < std::numeric_limits<double>::epsilon()  )
+  {
+    value = 0.0;
+  }
+  else if ( arg1 < 0.0 )
+  {
+    if ( std::fabs(arg2 - 0.0) < std::numeric_limits<double>::epsilon() )
+    {
+      value = 0.0;
+    }
+    else if ( arg2 < 0.0 )
+    {
+      value = 1.0;
+    }
+    else if ( 0.0 < arg2 )
+    {
+      value = -1.0;
+    }
+  }
+  else if ( 0.0 < arg1 )
+  {
+    if ( std::fabs(arg2 - 0.0) < std::numeric_limits<double>::epsilon() )
+    {
+      value = 0.0;
+    }
+    else if ( arg2 < 0.0 )
+    {
+      value = -1.0;
+    }
+    else if ( 0.0 < arg2 )
+    {
+      value = 1.0;
+    }
+  }
+
+  return value;
```

Comment:
```suggestion
  const auto sign1 = std::copysign(arg1, 1);
  const auto sign2 = std::copysign(arg2, 1);
  return sign1 * sign2;
```
Do remember to include `cmath` header for `copysign`

Reply:
changed (with reversed argument order of copysign)